layers/+lang/prolog/README.org

```diff
@@ -0,0 +1,92 @@
+#+TITLE: prolog layer
+
+# TOC links should be GitHub style anchors.
+* Table of Contents                                        :TOC_4_gh:noexport:
+- [[#description][Description]]
+  - [[#features][Features:]]
+- [[#install][Install]]
+- [[#key-bindings][Key bindings]]
+  - [[#leader][Leader]]
+    - [[#consulting][Consulting]]
+    - [[#compiling][Compiling]]
+    - [[#formatting][Formatting]]
+    - [[#inserting][Inserting]]
+    - [[#help][Help]]
+    - [[#evaluating][Evaluating]]
+
+* Description
+This layer adds support for Prolog using the bundled Prolog mode for Emacs. In addition it also adds ediprolog support for better interaction with SWI-Prolog. 
+
+** Features:
+  - Designed for SWI-Prolog as a default, but can be used with other Prologs that Prolog mode supports. 
+  - Interactive consulting and compiling.
+  - Auto-formatting.
+  - Apropos and help lookup.
+
+    
+* Install
+To use this configuration layer, add it to your =~/.spacemacs=. You will need to
+add =prolog= to the existing =dotspacemacs-configuration-layers= list in this
+file.
+
+Note that the layer currently uses the Prolog mode that is bundled with Emacs. 
+If you want to use the separately maintained [[https://bruda.ca/_media/emacs/prolog.el][Prolog Mode by Stefan D. Bruda]]:
+- Place =prolog.el= in the =local/prolog= directory relative the layer directory.
```

Comment:
I think requesting users to manually copy a package into a local folder is not a solution we can live with.
You should rather declare the package you wish to load in your `package.el` and provide an init function.

You could declare a layer variable and load the package only if it is set.
```
;; Would look like this
(setq auto-completion-packages
      '(
        company
        (company-quickhelp :toggle auto-completion-enable-help-tooltip)
        ))
```

Reply:
Using an external prolog-mode, as opposed to the built-in one, was an option I documented in the README that I tested previously. From using the layer I never had the need to use it over the prolog mode that comes as a default with emacs, which is the one currently integrated by default into the layer. I will remove the reference to the external layer from the documentation, as it functions just as well using the built-in prolog mode.

Reply:
Fine with me.